pkg/common/plugin/azure/msi_test.go

```diff
@@ -114,6 +109,63 @@ func TestFetchInstanceMetadata(t *testing.T) {
 	require.Equal(t, expected, metadata)
 }
 
+func TestMakeAgentID(t *testing.T) {
+	type args struct {
+		td                string
+		agentPathTemplate string
+		claims            *MSITokenClaims
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    string
+		wantErr bool
```

Comment:
instead of want error, can you instead validate the error you are getting? it is possible to get into situations where you get another error and not the expected one